test/mui_suggestors/components/shared_props_migrators/hit_area.dart

```diff
@@ -0,0 +1,336 @@
+import 'package:meta/meta.dart';
+import 'package:test/test.dart';
+
+import '../../../util.dart';
+import '../shared.dart';
+
+enum HitAreaMixinSkippableTests {
+  TYPE,
+}
+
+/// Shared tests for components that are migrating a component that mixes in
+/// `HitAreaMixin`.
+Future<void> sharedHitAreaMixinTests(
+    {@required String? startingFactoryName,
+    @required SuggestorTester? testSuggestor,
+    String? endingFactoryName,
+    String? extraEndingProps,
+    List<HitAreaMixinSkippableTests> testsToSkip = const []}) async {
+  if (endingFactoryName == null) {
+    endingFactoryName = startingFactoryName;
+  }
+
+  if (extraEndingProps == null) {
+    extraEndingProps = '';
+  }
+
+  if (startingFactoryName == null || testSuggestor == null) {
+    throw ArgumentError(
+        'startingFactoryName and testSuggestor are required parameters');
+  }
+
+  group('(shared `HitAreaMixin` tests)', () {
```

Comment:
These tests were mostly pulled from the button migrator test suite and then refactored a little to change the factory name and such